src/pages/OrderConfirmPage/OrderConfirmPage.tsx

```diff
@@ -1,27 +1,68 @@
-import { BottomButton } from '@components/common';
-import { useSelectedCartItems } from '@hooks/shoppingCart';
-import { totalPriceSelector } from '@recoil/shoppingCart';
-import { formatKoreanCurrency } from '@utils/index';
-import { useRecoilValue } from 'recoil';
-
-import * as Styled from './OrderConfirmPage.styled';
+import { PageExplanation } from '@components/common';
+import { PurchaseProcessLayout } from '@components/layout';
+import {
+  CouponModalContainer,
+  InvalidOrderAccessError,
+  OrderAmountsList,
+  SelectedItemList,
+  ShippingInfo,
+} from '@components/orderConfirm';
+import { useAvailableCoupons, useSelectedCartItems } from '@hooks/index';
+import useOrderCartItems from '@hooks/order/useOrderCartItems';
+import { availableCouponsAtom, totalPriceSelector } from '@recoil/shoppingCart';
+import { ROUTE_PATHS } from '@routes/route.constant';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
 
 const OrderConfirmPage: React.FC = () => {
+  const setAvailableCoupons = useSetRecoilState(availableCouponsAtom);
   const totalPrice = useRecoilValue(totalPriceSelector);
 
-  const { totalSelectedItemLength, selectedTotalQuantity } = useSelectedCartItems();
+  const { totalSelectedItemLength, selectedTotalQuantity, selectedItems } = useSelectedCartItems();
+  const { getAvailableCoupons } = useAvailableCoupons();
+  const { fetchError, orderCartItems } = useOrderCartItems();
+
+  const navigate = useNavigate();
+  const isValidAccess = selectedItems.length > 0;
+
+  const handleClickBottomButton = async () => {
```

Comment:
결제하기 버튼 클릭 시 다음을 진행해요.
1. order api 요청
2.  1의 요청이 설공하면 상태 초기화/리프레쉬
3.  navigate를 통해 결제 페이지에서 결제 정보를 넘여준다.

Reply:
> 구현 시에는 결제가 완료된 후에 보여지는 것이 결제 확인 페이지라서 결제로 넘어갈때 상태 초기화/리프레쉬를 해주는 것이 맞다 생각했어요. (중략) 제가 구현한 결제하기 버튼 이후의 상태처리에 대해 하루는 어떻게 생각하시는 지 궁금해요.

바다가 질문해주신 의도를 제가 잘 파악하지 못한 것 같아요 🥲 혹시 조금 더 부연설명 부탁드려도 될까요?

( orderCartItems

Reply:
결제가 완료되었을때, 장바구니,쿠폰등에 사용된 상태를  초기화 시키고 결제가 완료되어서 변경된 장바구니 목록 데이터를 새로 받아와서 업데이트 시켜야해요. 

초기화와 데이터 리프레시를 어디서/언제 진행하는 것이 그 목적에 더 맞을까가 궁금했어요.

저는 결제 확인 페이지는 이미 결제가 완료된 상태이기 때문에 저는 결제 확인 페이지에서는 장바구니 목록, 쿠폰등의 전역 상태는 초기화된다고 생각했어요. 그래서 결제하기 버튼을 클릭 했을때  결제하기 버튼을 클릭했을때 order api 요청이 성공하면 상태 초기화와 데이터 리프레시를 진행해요. 
하지만 저와 다른 타이밍에 상태 초기화/리프레쉬를 진행할 수 도 있어요. 예를 들면 A는 결제 확인 페이지인 컴포넌트가 마운트 되었을때 이를 진행할 수 있고, 또 다른 B는 결제 확인 페이지에서도 주문한 상품 수량, 결제 금액 데이터가 필요하니까 전역 상태를 살려두어서 이를 사용하고 결제 확인 페이지 컴포넌트의 언마운트 될때 상태 초기화와 장바구니 목록 데이터 리프레쉬를 진행할 수 도 있을 것 같아요.

Reply:
오 친절하고 상세하게 설명해주셔서 너무 감사합니다 덕분에 잘 이해했습니당 👍

> 1) 결제하기 버튼을 클릭했을때 order api 요청 성공 시
> 2) 결제 확인 페이지 컴포넌트 마운트 시
> 3) 결제 확인 페이지 컴포넌트 언마운트 시

로직 상 '초기화'는 결제 성공과 직접적으로 연결되어 있어보여요. 
따라서 1) 처럼 이 click 핸들러에서 담당해야 로직이 분산되지 않고 명확하게 상태관리가 된다고 느껴집니다.
특별히 결제 확인 페이지까지 전체 상태를 유지해야하는 이유가 없다면, 지금 구현해주신 방식이 가장 좋아보입니다.

<br/>

어떻게 보면, 로직을 이벤트핸들러에 넣어야할지, useEffect를 써야할지의 고민이라고도 볼 수 있을 것 같아요. 
관련해서 이 글을 추천드려봅니다.

> When you’re not sure whether some code should be in an Effect or in an event handler, ask yourself why this code needs to run. Use Effects only for code that should run because the component was displayed to the user. 

- 참고: https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers